spyder/plugins/editor/plugin.py

```diff
@@ -2284,6 +2273,85 @@ def new(self, fname=None, editorstack=None, text=None):
             if index is not None and not current_es.close_file(index):
                 return
 
+        # If no text is provided, create default content
+        try:
+            if text is None:
+                default_content = True
+                text, enc = encoding.read(self.TEMPLATE_PATH)
+                enc_match = re.search(r'-*- coding: ?([a-z0-9A-Z\-]*) -*-',
+                                      text)
+                if enc_match:
+                    enc = enc_match.group(1)
+
+                # Initialize template variables
+                if os.name == 'nt':
+                    # Windows
+                    import ctypes
+
+                    username = encoding.to_unicode_from_fs(
+                                os.environ.get('USERNAME', '-'))
+                    GetUserNameEx = ctypes.windll.secur32.GetUserNameExW
+                    name_display = 3
+
+                    size = ctypes.pointer(ctypes.c_ulong(0))
+                    GetUserNameEx(name_display, None, size)
+
+                    name_buffer = ctypes.create_unicode_buffer(
+                                size.contents.value)
+                    GetUserNameEx(name_display, name_buffer, size)
+                    displayname = name_buffer.value.strip()
+                else:
+                    # Linux, Mac OS X
+                    import pwd
+
+                    username = encoding.to_unicode_from_fs(
+                                   os.environ.get('USER', '-'))
+                    try:
+                        pwd_struct = pwd.getpwnam(username)
+                    except KeyError:
+                        displayname = ''
+                    else:
+                        # Use first element in Gecos field as user's full name
+                        displayname = pwd_struct.pw_gecos.split(',')[0].strip()
+
+                now = datetime.now().astimezone()
+                now.replace(microsecond=0)
+                VARS = {
+                    'date': now.ctime(),
+                    'isodate': now.isoformat(),
+                    'year': now.strftime('%Y'),
+                    'month': now.strftime('%m'),
+                    'day': now.strftime('%d'),
+                    'hour': now.strftime('%H'),
+                    'minute': now.strftime('%M'),
+                    'second': now.strftime('%S'),
+                    'tzname': now.strftime('%Z'),
+                    'monthname': now.strftime('%b'),
+                    'weekday': now.strftime('%a'),
+                    'username': username,
+                    'fullname': displayname or 'n/a',
```

Comment:
Here, I'd suggest just falling back to the username if there is no separate display name defined, as it seems much more sensible behavior for realistic use cases, and avoids the need to handle (and translate) a default.

```suggestion
                    'fullname': displayname or username,
```

Reply:
I also considered to use `username` as alternative, if `displayname` is not available. I decided against it, because I believed that it might come to a surprise, if one uses `fullname` and suddenly just the `displayname` is inserted. Maybe we should have `username`, `fullname` and `full_or_username` (or something similar)?